### PR TITLE
taito_b.cpp: DIP switch corrections for sbm & sbmj (MT08160)

### PR DESCRIPTION
**Pull request by scoomby**
The DIP switches have been corrected and tested as advised on MT08160.
Please let me know if anything needs to be changed or improved including DIP switch labelling.
Thanks in advance.

### CODE REVIEW

File: src/mame/drivers/taito_b.cpp [hunk 1]
```diff
@@ -1532,49 +1532,70 @@ static INPUT_PORTS_START( ryujin )
...
-	PORT_DIPNAME( 0x01, 0x01, DEF_STR( Unknown ) )      PORT_DIPLOCATION("SW1:1")
-	PORT_DIPSETTING(    0x01, DEF_STR( Off ))
-	PORT_DIPSETTING(    0x00, DEF_STR( On ))
-	PORT_DIPNAME( 0x02, 0x02, DEF_STR( Unknown ) )      PORT_DIPLOCATION("SW1:2")
-	PORT_DIPSETTING(    0x02, DEF_STR( Off ))
-	PORT_DIPSETTING(    0x00, DEF_STR( On ))
+	PORT_START("DSWA")
+	PORT_DIPNAME( 0x01, 0x01, "Tickets" )					PORT_DIPLOCATION("SW1:1")
+	PORT_DIPSETTING(    0x00, "Disabled" )
+	PORT_DIPSETTING(    0x01, "Enabled" )
```
Comment (Osso13) [inline src/mame/drivers/taito_b.cpp:1538]: In most of MAME "Disabled" and "Enabled" are standardized as DEF_STR( Off ) and DEF_STR( On ).

Comment (scoomby) [reply src/mame/drivers/taito_b.cpp]: Agreed

File: src/mame/drivers/taito_b.cpp [hunk 2]
```diff
@@ -1532,49 +1532,70 @@ static INPUT_PORTS_START( ryujin )
...
+		DIP 00: 0000 ;+00
+		DIP 01: FFFB ;-05
+		DIP 10: 0005 :+05
+		DIP 11: 000A :+10
+		For example, if the player had an initial power of 200t and Dip Switches 2-3 and 2-4 were both set, the power would be increased to 210t.
+		This bonus (or penalty) is applied before the game floors the punch power at 1t and before it caps the punch power at 300t. So, if in MAME, you hold Pad Photosensors 1 and 2 and then immediately hit Pad Photosensor 3, you'll still get a punch power of 300t even if Dip Switch 2-3 was set and Dip Switch 2-4 wasn't.
+		Just for reference, the initial power of the player's punch is calculated by taking 24000 and dividing it by the timer at 105048 in RAM. This is performed in the subroutine at 0x0A5B0 in the main CPU.
+	*/
+	PORT_DIPSETTING(    0x08, "Lower" )
+	PORT_DIPSETTING(    0x0c, "Medium" )
```
Comment (Osso13) [inline src/mame/drivers/taito_b.cpp:1571]: Sorry, just noticed this.
Medium, Higher and Highest also have standard DEF_STR( Medium ), DEF_STR( Higher ) and DEF_STR( Highest )

Comment (scoomby) [reply src/mame/drivers/taito_b.cpp:1571]: No problem at all.  Thanks for the comment

File: src/mame/drivers/taito_b.cpp [hunk 3]
```diff
@@ -1615,6 +1636,30 @@ static INPUT_PORTS_START( sbm )
...
+	PORT_DIPSETTING(    0x40, DEF_STR( 2C_1C ) )
+	PORT_DIPSETTING(    0x80, DEF_STR( 1C_2C ) )
+	PORT_DIPSETTING(    0xc0, DEF_STR( 1C_1C ) )
+
+	PORT_MODIFY("DSWB")
+	PORT_BIT( 0x10, IP_ACTIVE_LOW, IPT_UNKNOWN )
+	PORT_BIT( 0x20, IP_ACTIVE_LOW, IPT_UNKNOWN )
+	PORT_BIT( 0x40, IP_ACTIVE_LOW, IPT_UNKNOWN )
+	PORT_BIT( 0x80, IP_ACTIVE_LOW, IPT_UNKNOWN )
+INPUT_PORTS_END
```
Comment (cuavas) [inline src/mame/drivers/taito_b.cpp:1661]: This makes six of the DIP switches disappear.  The other DIP switches are physically present even if their purpose isn’t known.

Comment (scoomby) [reply src/mame/drivers/taito_b.cpp:1661]: Understood.

**Pull request merged**